upgrader.go

```diff
@@ -262,6 +262,20 @@ type neverCloseThisFile struct {
 
 func writePIDFile(path string) error {
 	dir, file := filepath.Split(path)
+
+	// if dir is empty, the user properly specified just the name
+	// of the pid file expecting it to be created in the current work directory
```

Comment:
Can you extend the comment to mention that this only works because `newOSProcess()` persists the original wd. Otherwise this could write to different directories.

Reply:
I didn't realize this - would it be too hacky to just use `initialWD` and then not have any errors to worry about?

Reply:
At that point getWD and initialWD have the same result I think. I think this might be a good idea actually, I'll check.

Reply:
I had a look, I think this should work. In this case it might make sense to panic if the call to GetWD to populate initialWD fails?